src/protocols/secure_channel/CASESession.cpp

```diff
@@ -264,22 +273,24 @@ CHIP_ERROR CASESession::EstablishSession(SessionManager & sessionManager, Fabric
 exit:
     if (err != CHIP_NO_ERROR)
     {
-        Clear();
+        mState = State::kError;
     }
+    VerifyOrDie(SanityCheck());
+    // EstablishSession is a API from delegate side, so we don't call OnSessionEstablishmentDone here, the delegate awares the error
+    // by return value.
     return err;
 }
 
 void CASESession::OnResponseTimeout(ExchangeContext * ec)
 {
-    VerifyOrReturn(ec != nullptr, ChipLogError(SecureChannel, "CASESession::OnResponseTimeout was called by null exchange"));
-    VerifyOrReturn(mExchangeCtxt == ec, ChipLogError(SecureChannel, "CASESession::OnResponseTimeout exchange doesn't match"));
-    ChipLogError(SecureChannel, "CASESession timed out while waiting for a response from the peer. Current state was %u", mState);
-    // Discard the exchange so that Clear() doesn't try closing it.  The
-    // exchange will handle that.
-    DiscardExchange();
-    Clear();
-    // Do this last in case the delegate frees us.
+    ChipLogError(SecureChannel, "CASESession timed out while waiting for a response from the peer. Current state was %u",
+                 static_cast<uint8_t>(mState));
+    AbortExchange();
```

Comment:
No, why are we doing AbortExchange here?  The exchange is in the middle of closing itself already; why would we abort it?

Reply:
There isn't too much difference between discard and abort here, the difference is whether clean pending mrp packets or not. Anyway, I'll change to discard to keep the change minimal.

Reply:
Conceptually, an abort here leads to a double-close.  The exchange happens to be guarding against that for now, but relying on those guards is not great.